_literate/EX-boston-flux.jl

```diff
@@ -0,0 +1,128 @@
+
+mutable struct MyNNRegressor <: MLJFlux.Builder
+    n1::Int #Number of cells in the first hidden layer
+    n2::Int #Number of cells in the second hidden layer
+end
+
+# Step 2: Building the neural network from this object.
+# Extend the `MLJFlux.fit` function. This takes in 3 arguments: The object of
+# `MyNNRegressor`, input dimension (ip) and output dimension (op).
+
+function MLJFlux.fit(model::MyNNRegressor, ip, op)
+    return Flux.Chain(Flux.Dense(ip, model.n1), Flux.Dense(model.n1, model.n2), Flux.Dense(model.n2, op))
+end
+
+# With all definitions ready, let us create an oject of this:
+
+myregressor = MyNNRegressor(20, 10)
+
+# Since the boston dataset is a regression problem, we'll be using
+# `NeuralNetworkRegressor` here. One thing to remember is that 
+# a `NeuralNetworkRegressor` object works seamlessly like any other
+# MLJ model: you can wrap it in an  MLJ `machine` and to anything
+# you'd do otherwise.
+
+# Let's start by defining our NeuralNetworkRegressor object, that takes `myregressor`
+# as it's parameter.
+
+nnregressor = NeuralNetworkRegressor(builder=myregressor, epochs=10)
+
+# Other parameters that `NeuralNetworkRegressor` takes in are:
+# optimiser = ADAM(0.001, (0.9, 0.999), IdDict{Any,Any}())
+# loss = Flux.mse,
+# epochs = 10,
+# batch_size = 1,
+# lambda = 0.0,
+# alpha = 0.0,
+# optimiser_changes_trigger_retraining = false
+
+# `nnregressor` now acts like any other MLJ model. Let's try wrapping it in a 
+# MLJ machine and calling `fit!, predict!`.
+
+mach = machine(nnregressor, features, targets)
+
+# Let's fit this on the train set
+
+fit!(mach, 3, rows=train, vebosity=3)
+
+# As we can see, the loss decreases at each epoch, showing the the neural network 
+# is gradually learning form the training set.
+
+predict(mach, rows=test)
+
+# Now let's retrain our model. One thing to remember is that retrainig may OR may not 
+# re-initialize our neural network model parameters. For example, changing the number of 
+# epochs to 15 will not causes the model to train to 15 epcohs, but just 5 additional
+# epochs.
+
+nnregressor.epochs = 15
+
+fit!(mach, rows=train, verbosity=3)
+
+# You can always specify that you want to retrain the model from scratch using the force=true
+# parameter. (Look at documentation for `fit!` for more).
+
+# However, changing parameters such as batch_size will necessarily cause re-training from scratch.
+nnregressor.batch_size = 2
+fit!(mach, rows=train, verbosity=3)
+
+# Another bit to remember here is that changing the optimiser doesn't cause retaining by default.
+# However, the `optimiser_changes_trigger_retraining` in NeuralNetworkRegressor can be toggled to
+# accomodate this.
+
+## Tuning
+
+# As mentioned above, `nnregressor` can act like any other MLJ model. Let's try to tune the
+# batch_size parameter.
+
+bs = range(nnregressor, :batch_size, lower=1, upper=5)
+
+tm = TunedModel(model=nnregressor, ranges=[bs, ], measure=Flux.mse)
+
```

Comment:
Unfortunately, I don't believe the Flux and MLJ API's for loss functions are compatible. See the note at the end of the readme. So here, you should use `measure=l2`, say.